src/config.rs

```diff
@@ -41,6 +41,10 @@ pub struct Config {
     // Gitlab authentication
     pub(crate) gitlab_accesstoken: Option<String>,
 
+    // request timeout in seconds
+    pub(crate) request_timeout: Option<u64>,
```

Comment:
Might be better to parse to `Option<Duration>` here. (It's currently a bit inconsistent, we have some other durations as integers, but also some other configs using semantic types).

Reply:
@Nemo157 you mean while keeping the format in the environment variable? 

I updated the code accordingly, you want to recheck?

Reply:
thinking about it, I'll merge it like this, we can optimize later. 

I want to get more grip on the current production issues

Reply:
Yeah, leaving it as seconds in the environment variable makes sense to me. We could use `humantime` to allow `DOCSRS_REPORT_REQUEST_TIMEOUT=1m`, but I don't think it's worth it.

Reply:
oh yeah, I was totally missing that we already have it in our dependencies :)